lib/client.js

```diff
@@ -254,16 +259,44 @@ Object.assign(Client.prototype, {
     // choose the smallest, positive timeout setting and set on poolConfig
     poolConfig.acquireTimeoutMillis = Math.min(...timeouts);
 
+    const updatePoolConnectionSettingsFromProvider = () => {
+      if (!this.connectionConfigProvider) {
+        return; // static configuration, nothing to update
+      }
+      if (
+        this.connectionSettings.expired === undefined ||
```

Comment:
Why not use falsiness check instead? If someone sets it to null explicitly, should be handled same way, and I assume that we don't want to support boolean values here.

Reply:
will change